examples/trainer/wscrossweigh/wscrossweigh_training_tutorial.py

```diff
@@ -28,38 +29,58 @@ def train_wscrossweigh(path_to_data: str, num_classes: int) -> None:
+
+    # Download data
+    client = Minio("knodle.dm.univie.ac.at", secure=False)
+    files = [
+        "df_train.csv", "df_dev.csv", "df_test.csv",
+        "train_rule_matches_z.lib", "dev_rule_matches_z.lib", "test_rule_matches_z.lib",
+        "mapping_rules_labels_t.lib"
+    ]
+    for file in tqdm(files):
+        client.fget_object(
+            bucket_name="knodle",
+            object_name=os.path.join("datasets/imdb/processed", file),
+            file_path=os.path.join(processed_data_dir, file),
+        )
+
+    # Load data into memory
+    df_train = pd.read_csv(os.path.join(processed_data_dir, "df_train.csv"))
+    df_dev = pd.read_csv(os.path.join(processed_data_dir, "df_dev.csv"))
+    df_test = pd.read_csv(os.path.join(processed_data_dir, "df_test.csv"))
+
+    mapping_rules_labels_t = joblib.load(os.path.join(processed_data_dir, "mapping_rules_labels_t.lib"))
+
+    train_rule_matches_z = joblib.load(os.path.join(processed_data_dir, "train_rule_matches_z.lib"))
+
+    # sample weights are calculated with logistic regression model (with TF-IDF features); the BERT model is used for
+    # the final classifier training.
+    train_tfidf_sparse, dev_tfidf_sparse, _ = get_tfidf_features(df_train["sample"].tolist(), df_dev["sample"].tolist())
     train_tfidf = Tensor(train_tfidf_sparse.toarray())
     train_dataset_tfidf = TensorDataset(train_tfidf)
 
-    # For the BERT training we convert train, dev and test data to BERT encoded features (namely, input indices and attention mask)
-    tokenizer = DistilBertTokenizer.from_pretrained('distilbert-base-uncased')
-    train_input_x_bert = get_bert_encoded_features(train_df["sample"].tolist(), tokenizer)
-    test_labels = TensorDataset(LongTensor(list(test_df.iloc[:, 1])))
-    test_dataset_bert = get_bert_encoded_features(test_df["sample"].tolist(), tokenizer)
+    # For the BERT training we convert train, dev, test data to BERT encoded features (input indices & attention mask)
+    model_name = 'distilbert-base-uncased'
+    tokenizer = DistilBertTokenizer.from_pretrained(model_name)
 
-    # for some datasets dev set is not provided. If it is not the case, is would be encoded with BERT features
-    # (since it is used only for final training) and passed to a class constructor or to train function
-    if dev_df is not None:
-        dev_dataset_bert = get_bert_encoded_features(dev_df["sample"].tolist(), tokenizer)
-        dev_labels = TensorDataset(LongTensor(list(dev_df.iloc[:, 1])))
-    else:
-        dev_labels, dev_dataset_bert = None, None
+    X_train = convert_text_to_transformer_input(df_train["sample"].tolist(), tokenizer)
+    X_dev = convert_text_to_transformer_input(df_dev["sample"].tolist(), tokenizer)
+    X_test = convert_text_to_transformer_input(df_test["sample"].tolist(), tokenizer)
+
+    y_dev = TensorDataset(LongTensor(list(df_dev.iloc[:, 1])))
+    y_test = TensorDataset(LongTensor(list(df_test.iloc[:, 1])))
```

Comment:
```suggestion
    y_dev = TensorDataset(LongTensor(df_dev.iloc[:, 1].tolist()))
    y_test = TensorDataset(LongTensor(df_test.iloc[:, 1].tolist()))
```

But also why not access the column by name with df["label"]?

Reply:
Ah apparantly somehow wasn't corrected last time we discussed it. done.